### PR TITLE
support parse-json-secrets

### PR DESCRIPTION
**Pull request by amirmao**
If the secret is json, create environment variables from the json secret for each key/value pair.

### CODE REVIEW

File: src/secrets.js [hunk 1]
```diff
@@ -180,4 +200,36 @@ function exportSecretToOutput(variableName, secretValue, exportSecretsToOutputs,
...
+    for (const [key, value] of Object.entries(parsedJson)) {
+        const upperCaseKey = key.toUpperCase();
+        if (keyMap.has(upperCaseKey)) {
+            throw new Error(`Duplicate key found in json: ${key}`);
+        }
+        keyMap.set(upperCaseKey, value);
+    }
+}
+
+function parseJson(jsonString) {
```
Comment (rani-sharim) [inline src/secrets.js:227]: I don't understand why hide the error here?

File: version [hunk 2]
```diff
@@ -1,2 +1,2 @@
 # Use Semantic versioning only. Please update the version number before opening a pull request.
-v1.0.1
+v1.0.2
```
Comment (rani-sharim) [inline version]: This isn't a patch, it's a minor (new feature)

File: src/secrets.js [hunk 3]
```diff
@@ -180,4 +200,36 @@ function exportSecretToOutput(variableName, secretValue, exportSecretsToOutputs,
...
+        prefix = convertPathNameToPrefix(secretName)
+    }
+    let outputName;
+    for (let key in parsedJson) {
+        outputName = prefix + "_" + key.toUpperCase()
+        exportSecretToOutput(outputName, parsedJson[key], exportSecretsToOutputs, exportSecretsToEnvironment)
+    }
+}
+
+function checkDuplicateKeys(parsedJson) {
```
Comment (rani-sharim) [inline src/secrets.js]: This is a bad function name imo. Implies it doesn't have side effects, but it does.

File: src/secrets.js [hunk 4]
```diff
@@ -180,4 +200,36 @@ function exportSecretToOutput(variableName, secretValue, exportSecretsToOutputs,
     }
 }
 
+function exportJsonFields(parsedJson, secretName, prefix, exportSecretsToOutputs, exportSecretsToEnvironment) {
+    if (prefix === undefined) {
```
Comment (rani-sharim) [inline src/secrets.js]: What about other values like `null`? do we need to handle them?

File: src/secrets.js [hunk 5]
```diff
@@ -142,13 +143,32 @@ async function exportPkiCertificateSecrets(akeylessToken, pkiCertificate, apiUrl
...
+        return
+    }
+
+    const secretValueOut = processSecretValue(secretValue, params['key']);
+    exportSecretToOutput(params['output-name'], secretValueOut, exportSecretsToOutputs, exportSecretsToEnvironment)
+}
+
+function convertPathNameToPrefix(path) {
+    // remove the first /
+    const pathWithoutFirstChar = path.slice(1)
```
Comment (rani-sharim) [inline src/secrets.js]: what if it doesn't start with /

File: src/secrets.js (hunk 1 above)
Comment (amirmao) [reply src/secrets.js:227]: Requirements, in case it is not a JSON, we will continue as usual

File: src/secrets.js (hunk 5 above)
Comment (amirmao) [reply src/secrets.js]: GP, I will add a validation for that

File: version (hunk 2 above)
Comment (amirmao) [reply version]: Done

File: src/secrets.js (hunk 3 above)
Comment (amirmao) [reply src/secrets.js]: Changed

File: src/secrets.js (hunk 4 above)
Comment (amirmao) [reply src/secrets.js]: Fixed

File: src/secrets.js (hunk 5 above)
Comment (amirmao) [reply src/secrets.js]: Done

**Review by rani-sharim — DISMISSED** [review]

File: dist/index.js [hunk 6]
```diff
@@ -225674,13 +225681,33 @@ async function exportPkiCertificateSecrets(akeylessToken, pkiCertificate, apiUrl
...
+function setOutput(secretValue, params, exportSecretsToOutputs, exportSecretsToEnvironment, parseJsonSecrets) {
+    if (parseJsonSecrets == true && !params.hasOwnProperty('key')) {
+        const parsedJson = parseJson(secretValue)
+        if (parsedJson != null) {
+            validateNoDuplicateKeys(parsedJson)
+            exportJsonFields(parsedJson, params['name'], params['prefix-json-secrets'], exportSecretsToOutputs, exportSecretsToEnvironment)
+        } else {
+            exportSecretToOutput(params['output-name'], secretValue, exportSecretsToOutputs, exportSecretsToEnvironment)
+        }
+        return
```
Comment (rani-sharim) [inline dist/index.js:225697]: Pointless `return`

File: dist/index.js [hunk 7]
```diff
@@ -225712,6 +225739,38 @@ function exportSecretToOutput(variableName, secretValue, exportSecretsToOutputs,
     }
 }
 
+function exportJsonFields(parsedJson, secretName, prefix, exportSecretsToOutputs, exportSecretsToEnvironment) {
+    if (!prefix) {
+        prefix = convertPathNameToPrefix(secretName)
+    }
+    let outputName;
```
Comment (rani-sharim) [inline dist/index.js:225746]: What's the point of defining this as `let` and outside the loop?

**Review by rani-sharim — DISMISSED** [review]

**Review by rani-sharim — APPROVED** [review]

**Review by avishay28 — APPROVED** [review]

**Pull request merged**